src/search/SearchBar.ts

```diff
@@ -115,7 +119,7 @@ export class SearchBar implements Component<SearchBarAttrs> {
 			placeholder: vnode.attrs.placeholder,
 			text: this.state().query,
 			busy: this.busy,
-			onInput: (text) => this.search(text),
+			onInput: debounceStart(300, (text) => this.search(text)),
```

Comment:
this does not work as you expect probably as you are creating the debounced function on every render. Besides, I thought there was already debounce elsewhere?